kodilla-patterns/src/main/java/com/kodilla/patterns/prototype/library/Library.java

```diff
@@ -0,0 +1,36 @@
+package com.kodilla.patterns.prototype.library;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Library extends Prototype {
+    private String name;
+    private Set<Book> books = new HashSet<>();
+
+    public Library(String name) {
+        this.name = name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public Set<Book> getBooks() {
+        return books;
+    }
+
+    public Library shallowCopy() throws CloneNotSupportedException {
+        return (Library) super.clone();
+    }
+
+    public Library deepCopy() throws CloneNotSupportedException {
+        Library clonedLibrary = (Library) super.clone();
+        clonedLibrary.books = new HashSet<>();
+        clonedLibrary.getBooks().addAll(books);
```

Comment:
czy to na pewno będzie kopia głęboka? Moim zdanie nie! W kursie jest pewna nieścisłość. 
Czy jak zmienię tytuł książki w bibliotece kopiowanej, to też chcę mieć zmieniony tytuł w drugiej bibliotece?